### DISCUSSION TITLE
What is the best way to filter based on a specific segment?

### DISCUSSION THREAD

**Discussion by dombarone** [nextgenhealthcare/connect#6154] [Q&A] [answered]
Good afternoon, 

I am looking to filter messages to different destination files if the msg['PID']['PID.3']['PID.3.1'].toString() contains only numeric values, and send all others that contain alphanumeric values to a separate destination file. My first thought was to use a filter in the destination tab of my channel but in the rule builder it doesn't look like I can use Regex. 

![image](https://github.com/nextgenhealthcare/connect/assets/152644451/a6077c7e-db12-4b38-a3ea-c4bde9e1e9d6)

What is the best way to accomplish this task? I'm sure I am missing something very simple, any help would be appreciated. 

Thanks, 

Dom

**Comment by pacmano1** [comment]
For the most part, the best way to filter is to use destinationSet.* filtering whereby you choose the destination on the _source_ side.

read up on:

![image](https://github.com/nextgenhealthcare/connect/assets/44065187/774d7b81-703d-45a8-bd96-4e1bda6fa169)

**Reply by pacmano1** [reply]
If you go over to the mirth forums there are many examples of regex use.

**Reply by pacmano1** [reply]
And if you care to read the manual on destination set filtering:

https://docs.nextgen.com/bundle/Mirth_User_Guide_4_5_0/page/connect/connect/topics/c_Use_Destination_Set_Filter_faq_mcug.html

**Reply by pacmano1** [reply, marked as answer]
And an example, but I don't use the UI tools, would just do a JavsScript step that does the same thing.

![image](https://github.com/nextgenhealthcare/connect/assets/44065187/13ff1639-6c28-434b-9daf-2080e482d63d)

**Reply by dombarone** [reply]
Thank you so much, I didn't realize that was an option on the source transformer, I agree with you, it is much easier to do this in JavaScript. This did the trick!
